packages/core/echo/echo-schema/src/types.ts

```diff
@@ -25,33 +52,22 @@ export interface Identifiable {
  */
 export type Ref<T> = T | undefined;
 
-// TODO(burdon): Name?
-export type EchoReactiveObject<T> = ReactiveObject<T> & Identifiable;
-
-export const foreignKey = (source: string, id?: string): ForeignKey => ({ source, id });
-
 /**
- * Reference to an object in a foreign database.
+ * Reactive object marker interface (does not change the shape of the object.)
+ * Accessing properties triggers signal semantics.
  */
-export type ForeignKey = {
-  /**
-   * Name of the foreign database/system.
-   * g. `github.com`.
-   */
-  source?: string;
-
-  /**
-   * Id within the foreign database.
-   */
-  id?: string;
-};
+export type ReactiveObject<T> = { [K in keyof T]: T[K] };
+
+export type EchoReactiveObject<T> = ReactiveObject<T> & Identifiable;
```

Comment:
Rename to just echo object?

Reply:
In a separate PR maybe 
<img width="328" alt="image" src="https://github.com/dxos/dxos/assets/9644546/43ed0798-17de-4736-9caf-2f1cd1b0fea1">